util/auto_skip_compressor.cc

```diff
@@ -65,27 +65,44 @@ Status AutoSkipCompressorWrapper::CompressBlock(
     return CompressBlockAndRecord(uncompressed_data, compressed_output,
                                   out_compression_type, wa);
   } else {
-    auto prediction = predictor_->Predict();
+    auto predictor_ptr =
+        static_cast<AutoSkipCompressionContext*>(wa->get())->predictor;
+    auto prediction = predictor_ptr->Predict();
     if (prediction <= kProbabilityCutOff) {
       // decide to compress
       return CompressBlockAndRecord(uncompressed_data, compressed_output,
                                     out_compression_type, wa);
     } else {
-      // decide to bypass compression
+      // bypassed compression
       *out_compression_type = kNoCompression;
       return Status::OK();
     }
   }
   return Status::OK();
 }
 
+Compressor::ManagedWorkingArea AutoSkipCompressorWrapper::ObtainWorkingArea() {
+  auto wrap_wa = wrapped_->ObtainWorkingArea();
+  return ManagedWorkingArea(
+      static_cast<WorkingArea*>(
+          new AutoSkipCompressionContext(std::move(wrap_wa))),
+      this);
+}
+void AutoSkipCompressorWrapper::ReleaseWorkingArea(WorkingArea* wa) {
+  delete static_cast<AutoSkipCompressionContext*>(wa);
+}
+
 Status AutoSkipCompressorWrapper::CompressBlockAndRecord(
     Slice uncompressed_data, std::string* compressed_output,
     CompressionType* out_compression_type, ManagedWorkingArea* wa) {
+  auto wrap_wa =
+      &(static_cast<AutoSkipCompressionContext*>(wa->get())->wrapped);
```

Comment:
Can't this function take AutoSkipCompressionContext instead of WorkingArea to avoid so many casts?